processor/filterprocessor/internal/common/processor.go

```diff
@@ -0,0 +1,211 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package common // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/filterprocessor/internal/common"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+	"go.uber.org/multierr"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/expr"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlresource"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlscope"
+)
+
+var _ baseContext = &resourceConditions{}
+
+type resourceConditions struct {
+	expr.BoolExpr[ottlresource.TransformContext]
+}
+
+func (r resourceConditions) Context() ContextID {
+	return Resource
+}
+
+func (r resourceConditions) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
+	var condErr error
+	td.ResourceSpans().RemoveIf(func(rspans ptrace.ResourceSpans) bool {
+		tCtx := ottlresource.NewTransformContext(rspans.Resource(), rspans)
+		condition, err := r.Eval(ctx, tCtx)
+		if err != nil {
+			condErr = multierr.Append(condErr, err)
+			return false
+		}
+		return condition
+	})
+	if td.ResourceSpans().Len() == 0 {
+		return processorhelper.ErrSkipProcessingData
+	}
+	return condErr
+}
+
+func (r resourceConditions) ConsumeMetrics(ctx context.Context, md pmetric.Metrics) error {
+	var condErr error
+	md.ResourceMetrics().RemoveIf(func(rmetrics pmetric.ResourceMetrics) bool {
+		tCtx := ottlresource.NewTransformContext(rmetrics.Resource(), rmetrics)
+		condition, err := r.Eval(ctx, tCtx)
+		if err != nil {
+			condErr = multierr.Append(condErr, err)
+			return false
+		}
+		return condition
+	})
+	if md.ResourceMetrics().Len() == 0 {
+		return processorhelper.ErrSkipProcessingData
+	}
+	return condErr
+}
+
+func (r resourceConditions) ConsumeLogs(ctx context.Context, ld plog.Logs) error {
+	var condErr error
+	ld.ResourceLogs().RemoveIf(func(rlogs plog.ResourceLogs) bool {
+		tCtx := ottlresource.NewTransformContext(rlogs.Resource(), rlogs)
+		condition, err := r.Eval(ctx, tCtx)
+		if err != nil {
+			condErr = multierr.Append(condErr, err)
+			return false
+		}
+		return condition
+	})
+	if ld.ResourceLogs().Len() == 0 {
+		return processorhelper.ErrSkipProcessingData
+	}
+	return condErr
+}
+
+var _ baseContext = &scopeConditions{}
+
+type scopeConditions struct {
+	expr.BoolExpr[ottlscope.TransformContext]
+}
+
+func (s scopeConditions) Context() ContextID {
+	return Scope
+}
+
+func (s scopeConditions) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
```

Comment:
It seems we've no coverage for the `scope` context (all signals). Can we please add some tests for that?